workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -14,76 +12,77 @@ public class SendCreatePlayerRequestSystem : ComponentSystem
+                var entities = initializationGroup.GetEntityArray();
+                for (var i = 0; i < entities.Length; ++i)
+                {
+                    PostUpdateCommands.AddComponent(entities[i], new ShouldRequestPlayerTag());
+                }
             }
 
-            for (var i = 0; i < sendData.Length; ++i)
+            if (!playerSpawnGroup.IsEmptyIgnoreFilter)
             {
-                var request = new CreatePlayerRequestType(new Improbable.Vector3f { X = 0, Y = 0, Z = 0 });
-                var createPlayerRequest = new PlayerCreator.CreatePlayer.Request(playerCreatorEntityId, request);
+                var entities = playerSpawnGroup.GetEntityArray();
+                for (var i = 0; i < entities.Length; ++i)
+                {
+                    var request = new CreatePlayerRequestType
+                    {
+                        Position = new Vector3f(0, 0, 0)
+                    };
+
+                    var createPlayerRequest = new PlayerCreator.CreatePlayer.Request(playerCreatorEntityId, request);
 
-                sendData.RequestSenders[i].RequestsToSend
-                    .Add(createPlayerRequest);
-                PostUpdateCommands.RemoveComponent<ShouldRequestPlayerTag>(sendData.Entities[i]);
+                    commandSystem.SendCommand(createPlayerRequest);
+                    PostUpdateCommands.RemoveComponent<ShouldRequestPlayerTag>(entities[i]);
+                }
             }
 
-            // Currently this has a race condition where you can receive two entites
-            // The fix for this is more sophisticted server side handling of requests
-            for (var i = 0; i < responseData.Length; ++i)
+            // Currently this has a race condition where you can receive two entities
+            // The fix for this is more sophisticated server side handling of requests
+            var responses = commandSystem.GetResponses<PlayerCreator.CreatePlayer.ReceivedResponse>();
+            foreach (var response in responses)
             {
-                foreach (var receivedResponse in responseData.Responses[i].Responses)
+                if (response.StatusCode == StatusCode.AuthorityLost)
                 {
-                    if (receivedResponse.StatusCode == StatusCode.AuthorityLost)
-                    {
-                        PostUpdateCommands.AddComponent(responseData.Entities[i], new ShouldRequestPlayerTag());
-                    }
-                    else if (receivedResponse.StatusCode != StatusCode.Success)
+                    if (workerSystem.TryGetEntity(response.EntityId, out var entity))
                     {
-                        logDispatcher.HandleLog(LogType.Error, new LogEvent(
-                            $"Create player request failed: {receivedResponse.Message}"));
+                        PostUpdateCommands.AddComponent(entity, new ShouldRequestPlayerTag());
```

Comment:
This is not the same.

Previously, it used the ECS entity that sent the command (the worker entity), now it uses the Entity Id of the entity which we _sent_ the command to.

Given the state of the world at startup - you will probably never have this entity checked out. 

This should instead add the tag to the worker entity again.